browser/src/control/Control.NavigatorPanel.ts

```diff
@@ -0,0 +1,186 @@
+/* -*- js-indent-level: 8 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+/*
+ * JSDialog.NavigatorPanel
+ */
+
+/* global app */
+class NavigatorPanel {
```

Comment:
Can we now create base class: `SidebarBase` which will be parent of Sidebar and Navigator.
We can then share almost all of the `onAdd`, `onRemove`, `isVisible`, `onJSUpdate`, `onJSAction`.
We only need to define strings which are different:
```
this.commandId = 'navigator' / 'sidebar';
this.containerWrapperId = 'navigator-dock-wrapper' / 'siebar-dock-wrapper';
this.panelId = 'navigator-panel' / 'sidebar-panel';
```

why do not concatenate even more:   `TYPE + '-dock-wrapper'` ? because it's then easier to `git grep` if id is fully present in the code and css to find related code :)